### PR TITLE
Create template and example for source

### PR DESCRIPTION
**Pull request by daceynolan**
To make all of the sources more uniform, I created a template and example file in the documentation for the team to look at when making source files. 

closes https://github.com/triggermesh/docs/issues/108

### CODE REVIEW

File: templates/source/sourceTemplate.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+## Deploying an Instance of the Source
+
+Open the Bridge creation screen and add a source of type **\_\_**
+
+Continue adding descriptions and steps needed.
+
+## Event Types
+
+Add the event type info here.
```
Comment (odacremolbap) [inline templates/source/sourceTemplate.md:35]:
```suggestion
Add the event type info here. If schemas are available, add link to those schemas for each type.
```
I think it helps people to know what the events flowing through the bridges look like when planning on connecting sources with targets, and target responses with other targets.

File: templates/source/sourceExample.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
+# Event source for AWS Cognito User Pool
+
+This event source captures messages from an [AWS Cognito User Pool ][cup-docs] whenever a specific action, such as the creation of a new user, happens in the user identity pool.
+
+## Prerequisites
+
+- AWS Cognito User Pool
+- Amazon Resource Name (ARN)
+- API Credentials
+
```
Comment (odacremolbap) [inline templates/source/sourceExample.md:10]: Maybe not an issue, but I find it hard and ugly the lack of indentation between ## and #####.
I think that is something that can be worked on the `mkdocs.yml` at the root. Should we create a followup to make this look nicer?

image

**Review by odacremolbap — APPROVED** [review]
Thanks for this!
+1 on my side.

Made a couple non blocking comments.
I also expect that if we agree on moving forward, the AWS Cognito example is removed from templates, replacing the actual AWS Cognito doc page.

Waiting for other people's view.

File: templates/source/sourceExample.md (hunk 2 above)
Comment (daceynolan) [reply templates/source/sourceExample.md:10]: Are you wanting this to be indented only are there other styles about it that you didn't like?

Comment (odacremolbap) [reply templates/source/sourceExample.md:10]: IMHO maybe adding weight - bolding the ## is enough.
It is still a bit subtle that transition between ## and ##### when prerequisites take more than one scroll page.

Comment (daceynolan) [reply templates/source/sourceExample.md:10]: Yeah, just when it was ### there was nothing differing the prerequisites with the sub information. THe API Credentials and ARN are supposed to be subheading of prerequisites. The main title is #. The main information is ##. And the sub info would be #####. It is just a part of the prerequisites so I needed to make the titles different.

File: templates/source/sourceTemplate.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+This event target receives .......
+
+## Prerequisites
+
+- unordered list out prerequisites
+- unordered list out prerequisites
+- unordered list out prerequisites
+
+##### This should be about a prerequisites
```
Comment (antoineco) [inline templates/source/sourceTemplate.md:11]: We go straight from h2 to h4, this is not great. If the problem is the styling applied by Mkdocs, we should fix the styling, not hack the h levels.

File: templates/source/sourceTemplate.md (hunk 1 above)
Comment (antoineco) [inline templates/source/sourceTemplate.md:35]: I think we need to templatize that section too.

I'd be in favor of something short and concise, like the event type emphasized (title and `backticks` maybe?) followed by a single sentence describing the event. Link to a schema whenever possible like Pablo said, this is critical IMO.

File: templates/source/sourceTemplate.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
+# Event Source for **\_\_\_\_**
+
+This event target receives .......
```
Comment (antoineco) [inline templates/source/sourceTemplate.md:3]: The title says "source", templates should be consistent.

File: templates/source/sourceTemplate.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+## Prerequisites
+
+- unordered list out prerequisites
```
Comment (antoineco) [inline templates/source/sourceTemplate.md:7]: Out -> Of (or did I get that wrong?)

File: templates/source/sourceTemplate.md (hunk 3 above)
Comment (daceynolan) [reply templates/source/sourceTemplate.md:11]: Oh sorry, I will override the styles. I didn't know that those "#" were the equivelent to h2 and h5. I do know how to make HTML semantic and use the correct ones for SEO. I had never really used markdown and didn't know what those #'s were for. Thanks for the review :)

File: templates/source/sourceTemplate.md (hunk 4 above)
Comment (daceynolan) [reply templates/source/sourceTemplate.md:3]: Yes, sorry. Targets were stuck on my brain. I will remove the "target" here.

File: templates/source/sourceTemplate.md (hunk 5 above)
Comment (daceynolan) [reply templates/source/sourceTemplate.md:7]: sorry yes should be of. I made edits to that and didn't fix that

File: templates/source/sourceTemplate.md (hunk 1 above)
Comment (daceynolan) [reply templates/source/sourceTemplate.md:35]: Okay I will write something up. Thanks for the help. That is why I asked for help creating these. I have no idea what info needs to be here and what does not  :)

Comment (daceynolan) [reply templates/source/sourceTemplate.md:35]: Okay I will write something up. Thanks for the help. That is why I asked for help creating these. I have no idea what info needs to be here and what does not :)

Comment (antoineco) [reply templates/source/sourceTemplate.md:35]: No need for a justification or apology 👍 Code reviews are for discussing changes, which is what we are doing here.

**Review by cab105 — APPROVED** [review]

**Review by odacremolbap — APPROVED** [review]

**Review by JeffNeff — APPROVED** [review]

**Pull request merged**